syntheseus/search/analysis/diversity.py

```diff
@@ -130,35 +132,32 @@ def _recursive_construct_packing_set(
         max_packing_number,
     )
     assert (
-        max_packing_number is None or len(route_set1) <= max_packing_number
+        max_packing_number is None or max(len(route_set1), len(route_set2)) <= max_packing_number
     ), "Max packing number exceeded in recursive call."
```

Comment:
Would it make sense to add a case here to directly return `route_set1` or `route_set2` if one of these sets has size `max_packing_number`? That way we can avoid the expensive computation below.

Reply:
Suggestion:

```python
# If max packing number is reached, directly return one set without merging
if len(route_set1) == max_packing_number:
    return route_set1
elif len(route_set2) == max_packing_number:
    return route_set2
```

Reply:
Even better: we can check the size of `route_set1` after the first recursive call, and if it's already reaching max value, return without doing the other call (see my last commit). This could potentially give a massive speedup if `max_packing_number` is much smaller than the number of routes.